### PR TITLE
Update the NFT Modal

### PR DESCRIPTION
**Pull request by muhadovic**
https://freedcamp.com/view/2956183/tasks/40073202

- Added Bid buy Modal
image

- Added bid offer modal
image

- Added deal approve modal
image

- Added review
image

- Added bid auction modal
image

- Added BidInput, ButtonAdd, ButtonClose, Link, Modal, Scrollbar Components.

### CODE REVIEW

File: src/common/components/Alert/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { ReactNode } from 'react'
```
Comment (dbizhev) [inline src/common/components/Alert/index.tsx:1]: @muhadovic change the file name to just Like "Alert.tsx"

File: src/common/components/BidInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/common/components/BidInput/index.tsx:1]: @muhadovic change the file name.

File: src/common/components/ButtonAdd/index.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/common/components/ButtonAdd/index.tsx:1]: @muhadovic change the file name.

File: src/common/components/ButtonClose/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/common/components/ButtonClose/index.tsx:1]: @muhadovic change the file name.

File: src/common/components/Link/index.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React, { ReactNode } from 'react'
```
Comment (dbizhev) [inline src/common/components/Link/index.tsx:1]: @muhadovic change the file name.

File: src/common/components/Modal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React, { ReactNode } from 'react'
```
Comment (dbizhev) [inline src/common/components/Modal/index.tsx:1]: @muhadovic change the file name.

File: src/common/components/Scrollbar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React, { ReactNode } from 'react'
```
Comment (dbizhev) [inline src/common/components/Scrollbar/index.tsx:1]: @muhadovic change the file name.

File: src/features/nft/bidAuctionModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/features/nft/bidAuctionModal/index.tsx:1]: @muhadovic change the file name

File: src/features/nft/bidBuyModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/features/nft/bidBuyModal/index.tsx:1]: Change the file name.

File: src/features/nft/bidOfferModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/features/nft/bidOfferModal/index.tsx:1]: @muhadovic change the file name

File: src/features/nft/dealApprovedModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/features/nft/dealApprovedModal/index.tsx:1]: @muhadovic change the file name.

File: src/features/nft/reviewModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
```
Comment (dbizhev) [inline src/features/nft/reviewModal/index.tsx:1]: @muhadovic change the file name

File: tailwind.config.js [hunk 13]
```diff
@@ -8,11 +8,11 @@ module.exports = {
...
-      sm: '900px',
-      md: '1100px',
-      lg: '1300px',
-      xl: '1440px',
-      '2xl': '1700px',
+      sm: '414px',
+      md: '834px',
+      lg: '874px',
+      xl: '946px',
+      '2xl': '1440px',
```
Comment (dbizhev) [inline tailwind.config.js:15]: @muhadovic why do you need those changes?

**Review by dbizhev — CHANGES_REQUESTED** [review]

File: tailwind.config.js (hunk 13 above)
Comment (muhadovic) [reply tailwind.config.js:15]: I think we have to imagine the small screens. But the sm in the current screen is 900px, it is so large value. Also, I think it is good to keep the tailwindcss default values. I don't understand why anyone set the large screen values.

File: src/features/nft/reviewModal/index.tsx (hunk 12 above)
Comment (muhadovic) [reply src/features/nft/reviewModal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/features/nft/bidBuyModal/index.tsx (hunk 9 above)
Comment (muhadovic) [reply src/features/nft/bidBuyModal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/features/nft/bidOfferModal/index.tsx (hunk 10 above)
Comment (muhadovic) [reply src/features/nft/bidOfferModal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/features/nft/dealApprovedModal/index.tsx (hunk 11 above)
Comment (muhadovic) [reply src/features/nft/dealApprovedModal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/Scrollbar/index.tsx (hunk 7 above)
Comment (muhadovic) [reply src/common/components/Scrollbar/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/features/nft/bidAuctionModal/index.tsx (hunk 8 above)
Comment (muhadovic) [reply src/features/nft/bidAuctionModal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/Modal/index.tsx (hunk 6 above)
Comment (muhadovic) [reply src/common/components/Modal/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/Link/index.tsx (hunk 5 above)
Comment (muhadovic) [reply src/common/components/Link/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/ButtonClose/index.tsx (hunk 4 above)
Comment (muhadovic) [reply src/common/components/ButtonClose/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/ButtonAdd/index.tsx (hunk 3 above)
Comment (muhadovic) [reply src/common/components/ButtonAdd/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/features/nft/bidBuyModal/index.tsx (hunk 9 above)
Comment (dbizhev) [reply src/features/nft/bidBuyModal/index.tsx:1]: @muhadovic rename it to BidOfferModal.tsx

File: src/common/components/Alert/index.tsx (hunk 1 above)
Comment (muhadovic) [reply src/common/components/Alert/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: src/common/components/BidInput/index.tsx (hunk 2 above)
Comment (muhadovic) [reply src/common/components/BidInput/index.tsx:1]: @dbizhev I follow the title of the design for filename. It it is wrong, please suggest the filename

File: tailwind.config.js (hunk 13 above)
Comment (dbizhev) [reply tailwind.config.js:15]: @muhadovic we agreed to screen sizes. We are all open for any suggestion. If you suggest to change the screen sizes, you should get approval from everyone. Ask everyone over Slack.

File: src/features/nft/bidBuyModal/index.tsx (hunk 9 above)
Comment (muhadovic) [reply src/features/nft/bidBuyModal/index.tsx:1]: @dbizhev , it is wrong. 

https://github.com/pastelnetwork/pastel-electron-wallet/pull/105
Please check 
image

File: tailwind.config.js (hunk 13 above)
Comment (dbizhev) [reply tailwind.config.js:15]: @muhadovic one more thing to note, you should consider this app is running on desktop, not mobile.

Comment (muhadovic) [reply tailwind.config.js:15]: @dbizhev  Yes, That is the desktop version. But we have to imagine when the users resize the app.

**Comment by dbizhev** [comment]
@muhadovic I don't see any component added to storybook. Add the components to storybook.

**Comment by muhadovic** [comment]
> @muhadovic I don't see any component added to storybook. Add the components to storybook.

I discussed this with Andres already.

**Comment by dbizhev** [comment]
> > @muhadovic I don't see any component added to storybook. Add the components to storybook.
> 
> I discussed this with Andres already.

This PR includes many modals. How to test those modals without storybook? And you should get approval from Jeff and Max.

**Comment by dbizhev** [comment]
@muhadovic I see you created the components per modal but we don't need it.

For example, the modals "buy it now" and "make an offer" are both similar styles, so we don't need to create two different components for those two modals. We do need only one component for those modals. Like this:

```
const Modal = ({
  type,
  contentA,
  contentB,
  title,
  description,
  buttonTitle,
  ...
}) => {
  if (type === "buy it now") {
    ...
  }

  if (type === "make an offer") {
    ...
  }
}
```

image

image

**Review by dbizhev — CHANGES_REQUESTED** [review]
@muhadovic please cleanup the modal components as I mentioned above.

**Comment by muhadovic** [comment]
> @muhadovic I see you created the components per modal but we don't need it.
> 
> For example, the modals "buy it now" and "make an offer" are both similar styles, so we don't need to create two different components for those two modals. We do need only one component for those modals. Like this:
> 
> ```
> const Modal = ({
>   type,
>   contentA,
>   contentB,
>   title,
>   description,
>   buttonTitle,
>   ...
> }) => {
>   if (type === "buy it now") {
>     ...
>   }
> 
>   if (type === "make an offer") {
>     ...
>   }
> }
> ```
> 
> image
> 
> image

@dbizhev I made the Modal component already. Please check it

**Pull request closed without merging**